pkg/kubectl/cmd/util/factory.go

```diff
@@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clientConfig = DefaultClientConfig(flags)
 	}
 
-	clients := NewClientCache(clientConfig)
+	clients := newClientCache(clientConfig)
```

Comment:
Then we really should either make clientCache public or make an interface and return that. I don't think NewClientCache should be public while returning a private type.